pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java

```diff
@@ -939,6 +940,70 @@ public Void call() throws Exception {
     }
   }
 
+  @Test
+  @Timeout(10)
+  void closeInProgressStatementProtocol32() throws Exception {
```

Comment:
Does the test differ from `closeInProgressStatement`? (looks like the only difference is 3.2 in the setup, however, I can't tell if the rest is the same or not)
If the core is the same, could you factor it to a say `private` method which you call from both of the tests?

It is hard to review and maintain code with large amount of duplicate logic.